swarm/src/dial_opts.rs

```diff
@@ -141,15 +147,23 @@ impl From<PeerId> for DialOpts {
 #[derive(Debug)]
 pub struct WithPeerId {
     peer_id: PeerId,
-    condition: PeerCondition,
+    condition: Option<PeerCondition>,
+    dial_conditions: DialConditions,
     role_override: Endpoint,
     dial_concurrency_factor_override: Option<NonZeroU8>,
 }
 
 impl WithPeerId {
     /// Specify a [`PeerCondition`] for the dial.
+    #[deprecated]
     pub fn condition(mut self, condition: PeerCondition) -> Self {
-        self.condition = condition;
+        self.condition = Some(condition);
+        self
+    }
+
+    /// Specify zero, one, or more [`DialConditions`]s for the dial.
```

Comment:
This needs to explain that these are conditions on why a dial should _not_ happen.